packages/graph/hash_graph/lib/graph/src/store/error.rs

```diff
@@ -74,6 +74,18 @@ impl fmt::Display for EntityDoesNotExist {
 
 impl Context for EntityDoesNotExist {}
 
+#[derive(Debug)]
+#[must_use]
+pub struct RaceConditionOnUpdate;
+
+impl fmt::Display for RaceConditionOnUpdate {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt.write_str("A race condition happened when updating the entity")
```

Comment:
Could this text error be more domain specific? Something about the entity being updated unexpectedly being changed?